cursorless-talon/src/spoken_forms.py

```diff
@@ -1,6 +1,6 @@
 import json
 from pathlib import Path
-from typing import Callable
+from typing import Callable, Concatenate, ParamSpec, TypeVar
```

Comment:
ParamSpec and Concatenate were new in Python 3.10. Did we intentionally drop support for Talon 0.3?

See [PEP 612](https://peps.python.org/pep-0612/)

Reply:
(I see our [pyproject.toml] still targets py39, so I'm guessing not 😬)

[pyproject.toml]: https://github.com/cursorless-dev/cursorless/blob/ae5304ddf3bd602ff2eda089a2fb1d5cddcf4e96/pyproject.toml

Reply:
Oh huh right good point. @AndreasArvidsson any thoughts on this one?

Reply:
Talon 0.3 was python 3.10 I believe? Anyhow community is going to move to Talon list so I don't think we need to support older Talon versions

Reply:
I still have Talon 0.3.1 on my laptops; it was Python 3.9.

Reply:
We were originally planning to wait for community to move to 0.4 before we did, but I don't think it's the end of the world to be slightly ahead of community here. From what I can tell, the delay on community side has more to do with ironing out details of upgrade PR than intentionally waiting for users to upgrade their talon version

That being said, we should prob bump our explicit requirement for Python to 3.10 if we're relying on 3.10 features